src/fmpz_mpoly/mpolyd.c

```diff
@@ -9,6 +9,12 @@
     (at your option) any later version.  See <https://www.gnu.org/licenses/>.
 */
 
+#ifdef __GNUC__
+# define printf __builtin_printf
+#else
+# include <stdio.h>
+#endif
+
```

Comment:
Why not change the printf to flint_printf?

Reply:
As the PR is pretty large, I want to keep a theme and try to leave the source code alone (I believe I changed some code in some header from printf to flint_printf in order to get rid of `stdio.h`).

I believe that all internals should use `printf`, but here I just didn't want to touch the code in the function.